### ISSUE TITLE
Fix storage permission for Android 11

### ISSUE THREAD

**Issue by 92lleo** [92lleo/WhatsappWebToGo#46]
To upgrade to and target Android 11 SDK, Storage permissions have to be adapted

**Comment by Minionguyjpro** [comment]
I'm on Android 11 on my tablet.

**Comment by 92lleo** [comment]
It works on android 11, but tooling is not at android 11 yet b/c of changes to storage permissions.

**Comment by somini** [comment]
> I don't think that the app-specific space is a good replacement as files won't show easy in your file explorer / when connecting via usb.

Personally, the only reason I use this app is to allow me to not give any permissions to the main WhatsApp app, and still send and receive media.
Requiring the storage permission on this is counter productive, might as well use the main app in that case, but if this is technically difficult I understand.

**Comment by 92lleo** [comment]
> Personally, the only reason I use this app is to allow me to not give any permissions to the main WhatsApp app, and still send and receive media. Requiring the storage permission on this is counter productive, might as well use the main app in that case, but if this is technically difficult I understand.

Understood. The storage permission is only required for downloading media to the device for sharing/storing etc. You can view media in the app also with storage permission not given (I guess you already know). Allowing download without a storage permission of any kind (there are special permissions in android 11+, which still need to be implemented) would require further logic and views in the app. I know the storage permission thing is not ideal currently, but it serves the download cause. 
In Addition: Giving storage permission to WAWTG is not the same as giving it to the native Whatsapp-App. The integrated whatsapp web still can't access the general storage, only the predefined "webview storage". Storage permission is only for the App wrapped around to be able to save things. It nevertheless requires additional trust.

-> still open

**Comment by somini** [comment]
> The storage permission is only required for downloading media to the device for sharing/storing etc. You can view media in the app also with storage permission not given (I guess you already know).

Yes, that's the main reason I use it, many thanks for releasing this.

> Allowing download without a storage permission of any kind (there are special permissions in android 11+, which still need to be implemented) would require further logic and views in the app.

Personally I think it's fine to just download the media to the app-specific folder, without the storage permission. 

I don't know enough about Android development, but I though the app-specific folder needed no storage permission? It's stuck on some weird `Android/data/...` folder, but at least you can access the files. I agree the UX is not very good, but it would be acceptable for me.
Generating a file name based on the current date is also fine.
